readers-frontend/src/pages/[year]/[month]/[day]/[slug].astro

```diff
@@ -0,0 +1,326 @@
+  // trying to find related ones.
+  const interimParentCategory = categories
+    .slice()
+    .reverse()
+    .find(
+      ({ category }) =>
+        (category.categories_nestUnder === null && // Nouse
+          category.categories_id !== 4) ||
+        category.categories_nestUnder === 4 // Muse
+    );
+
+  if (interimParentCategory) {
+    if (
+      // This is mainly for Muse articles as they are
+      // always going to have the Muse category listed
+      categories.length < 3 ||
+      categories.find(
+        ({ category }) =>
+          category.categories_nestUnder ===
+          interimParentCategory.category.categories_id
+      )
+    )
+      return interimParentCategory.category;
+  }
+
+  return categories[0].category;
+};
+
+export const dateFormatter = new Intl.DateTimeFormat("en-GB", {
+  year: "numeric",
+  month: "2-digit",
+  day: "2-digit",
+});
+
+export async function getStaticPaths() {
+  const articles = (await getAllArticles()).filter((article) => {
+    // Ensure article has an articleDraft
+    if (article.articlesDrafts.length === 0) return false;
+
+    // Ensure article has categories
+    if (article.categories.length === 0) return false;
+
+    // Some articles' slug is not formatted properly
+    const slug = article.articles_slug;
+    if (slug)
+      return (
+        !slug.includes("/") &&
+        !slug.includes("\\") &&
+        !slug.includes("#") &&
+        !slug.includes("?")
+      );
+
+    return false;
+  });
+
+  const similarArticles = await articles.reduce(
+    async (accumulatorPromise, article) => {
+      const accumulator = await accumulatorPromise;
+
+      const parentCategoryId = getParentCategory(
+        article.categories
+      ).categories_id;
+
+      // Get 4 most recent articles in parent category
+      const mSimilarArticlesIds = (
+        await prisma.articles.findMany({
+          where: {
+            articles_categories: {
+              contains: parentCategoryId.toString(),
+            },
+            articles_id: {
+              not: article.articles_id, // Exclude same article
+            },
+          },
+          take: 4,
+          select: {
+            articles_id: true,
+          },
+          orderBy: {
+            articles_published: "desc",
+          },
+        })
+      ).map((similarArticleId) => similarArticleId.articles_id); // Only take IDs
+
+      const mSimilarArticles = await getArticles(mSimilarArticlesIds);
+
+      accumulator.set(article.articles_id, mSimilarArticles);
+
+      return accumulator;
+    },
+    Promise.resolve(new Map<number, articlesWithArticleDrafts[]>())
+  );
+
+  return articles.map((article) => {
+    const date = article.articles_published || new Date(0);
+    // DD/MM/YYYY
+    const dateParts = dateFormatter.format(date).split("/");
+    return {
+      params: {
+        year: dateParts[2],
+        month: dateParts[1],
+        day: dateParts[0],
+        slug: article.articles_slug
```

Comment:
I don't think we should display articles without a slug

Reply:
That's a fair point tbh. How about we pass the articles with no slugs forward to the editors in the group chat and they can tell us what slugs to use? It would get around this issue

Reply:
Not sure if it's something they can fix by themselves on the backend tbh. There are also articles with invalid slugs. I found one with stuff like `?p=1666`. That would need a fix too.

Reply:
that's weird. yeah that would need fixing. I was thinking we would tell them what the articles are. They would tell us the slug to use. Finally we can do a migration to fix it.

Reply:
It would be outside the scope of this ticket ofc